`AbstractBaseSession` subclassing: Fields not overridable

# Bug report

Django allows [Extending database-backed session engines](https://docs.djangoproject.com/en/5.0/topics/http/sessions/#extending-database-backed-session-engines) ([example](https://docs.djangoproject.com/en/5.0/topics/http/sessions/#example)). Some of these fields need typings

## What's wrong

Fields of `AbstractBaseSession` (Django source: [5.0.6](https://github.com/django/django/blob/5.0.6/django/contrib/sessions/base_session.py), django-stubs: [5.0.0](https://github.com/typeddjango/django-stubs/blob/5.0.0/django-stubs/contrib/sessions/base_session.pyi#L13-L21)) are set to their return values instead of fields.

Consider the following Custom Session Backend, extending `session_key`'s `max_length`:

```python
from django.contrib.sessions.base_session import AbstractBaseSession
from django.db import models
from django.utils.translation import gettext_lazy as _


class CustomServiceSession(AbstractBaseSession):
    session_key = models.CharField(
        _("session key"),
        max_length=512,
        primary_key=True,
    )
```

This raises:

```
example.py:7: error: Incompatible types in assignment (expression has type "CharField[str | int | Combinable, str]", base class "AbstractBaseSession" defined the type as "str")  [assignment]
```

## How is that should be

`session_key` and other `AbstractBaseSession` fields should be overridable as model fields.

Perhaps something similar to `AbstractBaseUser` (django: [5.0.6](https://github.com/django/django/blob/5.0.6/django/contrib/auth/base_user.py#L60-L61), django-stubs: [5.0.0](https://github.com/typeddjango/django-stubs/blob/5.0.0/django-stubs/contrib/auth/base_user.pyi#L16-L21))

## System information

- OS: mac / linux
- `python` version: 3.10.1
- `django` version: 4.2
- `mypy` version: 1.9.0
- `django-stubs` version: 5.0.0
- `django-stubs-ext` version: 0.7.0+

--- a/django-stubs/contrib/sessions/base_session.pyi
+++ b/django-stubs/contrib/sessions/base_session.pyi
@@ -11,9 +11,9 @@ class BaseSessionManager(models.Manager[_T]):
     def save(self, session_key: str, session_dict: dict[str, Any], expire_date: datetime) -> _T: ...
 
 class AbstractBaseSession(models.Model):
-    expire_date: datetime
-    session_data: str
-    session_key: str
+    session_key = models.CharField(primary_key=True)
+    session_data = models.TextField()
+    expire_date = models.DateTimeField()
     objects: Any
 
     @classmethod

--- a/django-stubs/contrib/sessions/base_session.pyi
+++ b/django-stubs/contrib/sessions/base_session.pyi
@@ -1,8 +1,9 @@
 from datetime import datetime
-from typing import Any, TypeVar
+from typing import Any, ClassVar, TypeVar
 
 from django.contrib.sessions.backends.base import SessionBase
 from django.db import models
+from typing_extensions import Self
 
 _T = TypeVar("_T", bound=AbstractBaseSession)
 
@@ -14,6 +15,7 @@ class AbstractBaseSession(models.Model):
     session_key = models.CharField(primary_key=True)
     session_data = models.TextField()
     expire_date = models.DateTimeField()
+    objects: ClassVar[BaseSessionManager[Self]]
 
     @classmethod
     def get_session_store_class(cls) -> type[SessionBase] | None: ...

--- a/django-stubs/contrib/sessions/base_session.pyi
+++ b/django-stubs/contrib/sessions/base_session.pyi
@@ -14,7 +14,6 @@ class AbstractBaseSession(models.Model):
     session_key = models.CharField(primary_key=True)
     session_data = models.TextField()
     expire_date = models.DateTimeField()
-    objects: Any
 
     @classmethod
     def get_session_store_class(cls) -> type[SessionBase] | None: ...

--- a/scripts/stubtest/allowlist_todo.txt
+++ b/scripts/stubtest/allowlist_todo.txt
@@ -509,6 +509,7 @@ django.contrib.sessions.backends.signed_cookies.SessionStore.exists
 django.contrib.sessions.base_session.AbstractBaseSession.expire_date
 django.contrib.sessions.base_session.AbstractBaseSession.get_next_by_expire_date
 django.contrib.sessions.base_session.AbstractBaseSession.get_previous_by_expire_date
+django.contrib.sessions.base_session.AbstractBaseSession.objects
 django.contrib.sessions.base_session.AbstractBaseSession.session_data
 django.contrib.sessions.base_session.AbstractBaseSession.session_key
 django.contrib.sessions.base_session.BaseSessionManager.__slotnames__

